tools/rosnode/src/rosnode/__init__.py

```diff
@@ -808,9 +808,12 @@ def rosnodemain(argv=None):
             _fullusage()
     except socket.error:
         print("Network communication failed. Most likely failed to communicate with master.", file=sys.stderr)
+        sys.exit(1)
```

Comment:
My preferred style would be to `return 1` here, and then `sys.exit(rosnode.rosnodemain())` in the wrapper script, but there are direct sys.exit calls elsewhere in this same function, so I think it makes sense to follow the prevailing style as you have done.